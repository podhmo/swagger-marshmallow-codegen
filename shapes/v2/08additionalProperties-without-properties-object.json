{
  "definitions": {
    "D": {
      "additionalProperties": {
        "properties": {
          "name": {
            "type": "string"
          }
        },
        "type": "object"
      }
    },
    "O": {
      "properties": {
        "name": {
          "type": "string"
        }
      },
      "type": "object"
    },
    "S": {
      "properties": {
        "x": {
          "additionalProperties": {
            "properties": {
              "name": {
                "type": "string"
              }
            },
            "type": "object"
          }
        },
        "y": {
          "additionalProperties": {
            "$ref": "#/definitions/O"
          }
        },
        "z": {
          "$ref": "#/definitions/D"
        }
      },
      "type": "object"
    }
  },
  "verssion": "2.0.0"
}
