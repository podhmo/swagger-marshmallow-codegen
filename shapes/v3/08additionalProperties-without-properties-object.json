{
  "components": {
    "schemas": {
      "D": {
        "additionalProperties": {
          "properties": {
            "name": {
              "type": "string"
            }
          },
          "type": "object"
        }
      },
      "O": {
        "properties": {
          "name": {
            "type": "string"
          }
        },
        "type": "object"
      },
      "S": {
        "properties": {
          "x": {
            "additionalProperties": {
              "properties": {
                "name": {
                  "type": "string"
                }
              },
              "type": "object"
            }
          },
          "y": {
            "additionalProperties": {
              "$ref": "#/components/schemas/O"
            }
          },
          "z": {
            "$ref": "#/components/schemas/D"
          }
        },
        "type": "object"
      }
    }
  },
  "openapi": "3.1.0"
}
