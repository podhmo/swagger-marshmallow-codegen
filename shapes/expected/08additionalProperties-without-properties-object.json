from marshmallow import (
    Schema,
    fields,
)
from swagger_marshmallow_codegen.schema import AdditionalPropertiesSchema


class D(AdditionalPropertiesSchema):

    class Meta:
        additional_field = fields.Field()



class O(Schema):
    name = fields.String()


class S(Schema):
    x = fields.Dict(keys=fields.String(), values=fields.Nested())
    y = fields.Dict(keys=fields.String(), values=fields.Nested())
    z = fields.Dict(keys=fields.String(), values=fields.Nested())
